proxy/proxy/client.go

```diff
@@ -489,6 +489,11 @@ func (c *Client) InstanceVersionContext(ctx context.Context, instance string) (s
 	return version, nil
 }
 
+// AvailableConn returns false if MaxConnections is set and has been reached, true otherwise.
+func (c *Client) AvailableConn() bool {
+	return c.MaxConnections == 0 || atomic.LoadUint64(&c.ConnectionsCounter) < c.MaxConnections
```

Comment:
Can we drop `c.MaxConnections == 0 ||` from this statement?

Reply:
Correct me If I were wrong -

I think they are trying to include the cases that max_conn is not set

	// Settings for limits
	maxConnections = flag.Uint64("max_connections", 0,
		`If provided, the maximum number of connections to establish before refusing
new connections. Defaults to 0 (no limit)`,

Reply:
Yes, you're correct. Good catch.